velox/functions/sparksql/aggregates/tests/SumAggregationTest.cpp

```diff
@@ -38,5 +118,308 @@ TEST_F(SumAggregationTest, hookLimits) {
   testHookLimits<int64_t, int64_t, true>();
 }
 
+TEST_F(SumAggregationTest, decimalSumCompanionPartial) {
+  std::vector<int64_t> shortDecimalRawVector;
+  int128_t sum = 0;
+  for (int i = 0; i < 100; ++i) {
+    shortDecimalRawVector.emplace_back(i * 1000);
+    sum += i * 1000;
+  }
+
+  auto input = makeRowVector(
+      {makeFlatVector<int64_t>(shortDecimalRawVector, DECIMAL(10, 1))});
+  auto plan = PlanBuilder()
+                  .values({input})
+                  .singleAggregation({}, {"spark_sum_partial(c0)"})
+                  .planNode();
+  std::vector<int128_t> sumVector = {sum};
+  std::vector<bool> isEmptyVector = {false};
+  auto expected = makeRowVector({makeRowVector(
+      {makeFlatVector<int128_t>(sumVector, DECIMAL(20, 1)),
+       makeFlatVector<bool>(isEmptyVector)})});
+  AssertQueryBuilder(plan).assertResults(expected);
+}
+
+TEST_F(SumAggregationTest, decimalSumCompanionMerge) {
+  auto intermediateInput = makeRowVector({makeRowVector(
+      {makeFlatVector<int128_t>(
+           std::vector<int128_t>{1000, 2000, 3000}, DECIMAL(20, 1)),
+       makeFlatVector<bool>(std::vector<bool>{false, false, false})})});
+
+  auto plan = PlanBuilder()
+                  .values({intermediateInput})
+                  .singleAggregation({}, {"spark_sum_merge(c0)"})
+                  .planNode();
+  auto expected = makeRowVector({makeRowVector(
+      {makeFlatVector<int128_t>(std::vector<int128_t>{6000}, DECIMAL(20, 1)),
+       makeFlatVector<bool>(std::vector<bool>{false})})});
+  AssertQueryBuilder(plan).assertResults(expected);
+}
+
+TEST_F(SumAggregationTest, decimalSum) {
+  std::vector<std::optional<int64_t>> shortDecimalRawVector;
+  std::vector<std::optional<int128_t>> longDecimalRawVector;
+  for (int i = 0; i < 1000; ++i) {
+    shortDecimalRawVector.emplace_back(i * 1000);
+    longDecimalRawVector.emplace_back(HugeInt::build(i * 10, i * 100));
+  }
+  shortDecimalRawVector.emplace_back(std::nullopt);
+  longDecimalRawVector.emplace_back(std::nullopt);
+  auto input = makeRowVector(
+      {makeNullableFlatVector<int64_t>(shortDecimalRawVector, DECIMAL(10, 1)),
+       makeNullableFlatVector<int128_t>(longDecimalRawVector, DECIMAL(23, 4))});
+  createDuckDbTable({input});
+  testAggregations(
```

Comment:
Could you use the `AggregationTestBase::testAggregationsWithCompanion` API to test companion functions together with each testAggregations call as well? There are examples in AverageAggregationTest.cpp and ArrayAggTest.cpp.